internal/k8s/controller.go

```diff
@@ -1400,6 +1400,73 @@ func (lbc *LoadBalancerController) emitEventForVirtualServers(eventType string,
 	}
 }
 
+func getStatusFromEventTitle(eventTitle string) string {
+	switch eventTitle {
+	case "AddedOrUpdatedWithError", "Rejected", "NoVirtualServersFound", "Missing Secret", "UpdatedWithError":
+		return conf_v1.StateInvalid
+	case "AddedOrUpdatedWithWarning", "UpdatedWithWarning":
+		return conf_v1.StateWarning
+	case "AddedOrUpdated", "Updated":
+		return conf_v1.StateValid
+	default:
+		return "Unknown"
+	}
+}
+
+func (lbc *LoadBalancerController) updateVirtualServerVirtualServerRouteStatusesFromEvents() error {
+	vss := lbc.getVirtualServers()
+	var allErrs []error
+	for _, vs := range vss {
+		events, err := lbc.client.CoreV1().Events(vs.Namespace).List(context.TODO(),
+			meta_v1.ListOptions{FieldSelector: fmt.Sprintf("involvedObject.name=%v,involvedObject.uid=%v", vs.Name, vs.UID)})
+		if err != nil {
+			allErrs = append(allErrs, fmt.Errorf("error trying to get events for VirtualServer %v/%v: %v", vs.Namespace, vs.Name, err))
+			break
+		}
+
+		var timestamp time.Time
+		var latestEvent api_v1.Event
+		for _, event := range events.Items {
```

Comment:
what if the list of events is `0`. in that case latestEvent will be an empty struct and `UpdateVirtualServerStatus` will update the resource with `Unknown` state. Is it intentional? Or should we skip the update in that case?

same for VSRs

Reply:
It was intentional as we wanted to have at least something in the `status` field ASAP, but I can skip the update for sure. Either works fine